client/src/pages/game_lobby/TeamSelect.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import AddCircle from "@material-ui/icons/AddCircle";
+import Grid from "@material-ui/core/Grid";
+import IconButton from "@material-ui/core/IconButton";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
+import RemoveCircle from "@material-ui/icons/RemoveCircle";
+import Typography from "@material-ui/core/Typography";
+
+import { TEAM_CODE, DEFAULT_TEAM_STATE } from "./TeamPresets";
+import "./TeamSelect.css";
+
+const DEFAULT_RED_TEAM_STATE = JSON.parse(JSON.stringify(DEFAULT_TEAM_STATE));
```

Comment:
Hi, I'm looking through the code and trying to understand it. Could you explain a bit why we need to stringigy first and parse it after. Thanks.

Reply:
Originally I wanted a deep copy of DEFAULT_TEAM_STATE with:
const DEFAULT_RED_TEAM_STATE = [...DEFAULT_TEAM_STATE];
const DEFAULT_BLUE_TEAM_STATE = [...DEFAULT_TEAM_STATE];

but when I set both DEFAULT_RED_TEAM_STATE and DEFAULT_BLUE_TEAM_STATE on useState(), I was changing both redTeam and blueTeam arrays at the same time with the same data which leads me to believe that React thinks redTeam and blueTeam states are both referencing the same array when it shouldn't be the case. I read online that the stringify and parse method makes it so that it's absolutely a deep copy and when I tried it, it fixed the issue. 

I've just started using React Hooks so I will continue reading up more on it. Let me know if there's a better way around this. I could've easily just hardcored the JSON into both of those constants but I wanted to minimize duplicate code.

Reply:
Oh, That's quite wired. I'm also not sure how to tackle this unfortunately. Thanks for the thorough explanation.